app/controllers/user_subscriptions_controller.rb

```diff
@@ -0,0 +1,87 @@
+class UserSubscriptionsController < ApplicationController
+  before_action :authenticate_user!
+
+  def subscribed
+    params.require(%i[source_type source_id])
+    source_type = params[:source_type]
+    source_id = params[:source_id]
+
+    is_subscribed = UserSubscriptions::CacheChecker.new(current_user, source_type, source_id).cached_subscription_check
```

Comment:
Do we want to follow the convention we use elsewhere here and add a class level `.call` method?

```suggestion
    is_subscribed = UserSubscriptions::CacheChecker.call(current_user, source_type, source_id)
```

The method can be as simple as 

```ruby
def self.call(user, source_type, source_id)
  new(user, source_type, source_id).cached_subscription_check
end
```

Reply:
I went to do that originally and decided against it to leave room for more methods in `CacheChecker`. But on second thought, I can rename it to `SubscribtionCacheChecker` and implement this suggestion 👍